lib/superform/rails/components/select.rb

```diff
@@ -2,22 +2,55 @@ module Superform
   module Rails
     module Components
       class Select < Field
-        def initialize(*, collection: [], **, &)
+        def initialize(
+          *,
+          options: [],
+          collection: nil,
+          multiple: false,
+          **,
+          &
+        )
           super(*, **, &)
-          @collection = collection
+
+          # Handle deprecated collection parameter
+          if collection && options.empty?
+            warn "[DEPRECATION] Superform::Rails::Components::Select: " \
+                 "`collection:` keyword is deprecated and will be removed. " \
+                 "Use positional arguments instead: field.select([1, 'A'], [2, 'B'])"
+            options = collection
+          end
+
+          @options = options
+          @multiple = multiple
         end
 
-        def view_template(&options)
+        def view_template(&block)
+          # Hidden input ensures a value is sent even when all options are
+          # deselected in a multiple select
+          if @multiple
+            hidden_name = field.parent.is_a?(Superform::Field) ? dom.name : "#{dom.name}[]"
+            input(type: "hidden", name: hidden_name, value: "")
+          end
+
           if block_given?
-            select(**attributes, &options)
+            select(**attributes, &block)
           else
-            select(**attributes) { options(*@collection) }
+            select(**attributes) do
+              # If first option is nil, render a blank option
+              include_blank = @options.first.nil?
+              filtered_options = include_blank ? @options[1..] : @options
+
+              blank_option if include_blank
+              options(*filtered_options)
+            end
           end
         end
 
         def options(*collection)
           map_options(collection).each do |key, value|
-            option(selected: field.value == key, value: key) { value }
+            # Handle both single values and arrays (for multiple selects)
+            selected = Array(field.value).include?(key)
```

Comment:
This allocates a new array per iteration. Move `selected` outside of the loop if it doesn't need to be there and then check if the key is included in `selected`.

Reply:
Done